internal/telemetry/tracker_test.go

```diff
@@ -0,0 +1,104 @@
+package telemetry
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"gopkg.in/segmentio/analytics-go.v3"
+)
+
+type testClient struct {
+	calls []interface{}
+}
+
+func (client *testClient) Enqueue(track analytics.Message) error {
+	client.calls = append(client.calls, track)
+	return nil
+}
+
+func (client *testClient) Close() error {
+	return nil
+}
+
+const (
+	testID          = "id123"
+	testUserID      = "user123"
+	testExecutionID = "execution123"
+)
+
+var testTime = time.Date(2021, 1, 2, 3, 4, 5, 6, time.UTC)
+
+func TestNoopTracker(t *testing.T) {
+	t.Run("should create a noop tracker and should not do anything when track is called", func(t *testing.T) {
+		tracker := noopTracker{}
+		testEvent := createEvent(EventTypeCommandStart, nil, "someCommand")
+		testTrackStdoutOutput(t, &tracker, testEvent, "")
+	})
+}
+
+func TestStdoutTracker(t *testing.T) {
+	t.Run("should create an stdout tracker and should print the tracking information to stdout", func(t *testing.T) {
+		tracker := stdoutTracker{}
+		testTrackStdoutOutput(
+			t,
+			&tracker,
+			createEvent(EventTypeCommandComplete, nil, "someCommand"),
+			"03:04:05 UTC TELEM someCommand: COMMAND_COMPLETE[]\n",
+		)
+	})
+}
+
+func TestSegmentTracker(t *testing.T) {
+	t.Run("should create the segment tracker and should print the tracking information to the logger", func(t *testing.T) {
+		client := &testClient{}
+		tracker := segmentTracker{}
+		tracker.client = client
+		tracker.Track(createEvent(EventTypeCommandError, []EventData{{Key: EventDataKeyErr, Value: "Something"}}, "someCommand"))
+
+		expectedTrack := analytics.Track{
+			MessageId: testID,
+			UserId:    testUserID,
+			Timestamp: testTime,
+			Event:     string(EventTypeCommandError),
+			Properties: map[string]interface{}{
+				EventDataKeyErr:         "Something",
+				eventDataKeyCommand:     "someCommand",
+				eventDataKeyExecutionID: testExecutionID,
+			},
+		}
+		assert.Equal(t, []interface{}{expectedTrack}, client.calls)
+	})
```

Comment:
not sure why i missed this until now, but we want a test here to verify the error case works as expected as well

basically, the same approach you took with `testClient`, but instead of capturing `client.calls` you'll just return an error.

while you add this, lets also rework the names a bit to avoid any ambiguity/confusion.  the `testClient` you currently have should be come `mockSegmentClient`, and this new one can be `mockFailureSegmentClient`